packages/react/src/components/Popup/Popup.tsx

```diff
@@ -534,6 +569,33 @@ export default class Popup extends AutoControlledComponent<PopupProps, PopupStat
       ? activeElement
       : this.triggerRef.current
   }
+
+  updateContextPosition(nativeEvent: MouseEvent) {
+    if (!nativeEvent || nativeEvent.which !== 3) {
+      this.rightClickReferenceObject = null
+      return
+    }
+
+    const left = nativeEvent.clientX
+    const top = nativeEvent.clientY
+    const right = left + 1
+    const bottom = top + 1
+
+    function getBoundingClientRect() {
+      return {
+        left,
+        top,
+        right,
+        bottom,
+      }
+    }
+
+    this.rightClickReferenceObject = {
+      getBoundingClientRect,
+      clientWidth: 1,
+      clientHeight: 1,
+    }
```

Comment:
I believe that this logic can be extracted to a function like `createReferenceFromContextClick(e)`

Reply:
do you mean to move it outside of this file?

Reply:
Outside component at least...

Reply:
+1 for that, it can be in the same folder `packages/react/src/components/Popup/createReferenceFromContextClick.ts`